src/input.rs

```diff
@@ -32,6 +32,27 @@ pub struct Events<R> {
     leftover: Option<u8>,
 }
 
+/// An iterator that takes a single value from an option.
+struct OptionIterator<'a, T: 'a> {
```

Comment:
I wished something like that were available in the std lib, but `std::option::Iter` is not quite what we want, as we can't easily keep the `Option` if `next()` is never called.